dotnet/src/Connectors/Connectors.AzureOpenAI/Core/AzureClientCore.cs

```diff
@@ -122,12 +126,25 @@ internal AzureClientCore(
     /// <param name="httpClient">Custom <see cref="HttpClient"/> for HTTP requests.</param>
     /// <param name="serviceVersion">Optional API version.</param>
     /// <returns>An instance of <see cref="OpenAIClientOptions"/>.</returns>
-    internal static AzureOpenAIClientOptions GetAzureOpenAIClientOptions(HttpClient? httpClient, AzureOpenAIClientOptions.ServiceVersion? serviceVersion = null)
+    internal static AzureOpenAIClientOptions GetAzureOpenAIClientOptions(HttpClient? httpClient, string? serviceVersion = null)
     {
-        AzureOpenAIClientOptions options = serviceVersion is not null
-            ? new(serviceVersion.Value) { UserAgentApplicationId = HttpHeaderConstant.Values.UserAgent }
-            : new() { UserAgentApplicationId = HttpHeaderConstant.Values.UserAgent };
+        AzureOpenAIClientOptions.ServiceVersion? sdkVersion = null;
+        if (serviceVersion is not null)
+        {
+            sdkVersion = serviceVersion switch // Azure SDK versioning
+            {
+                "2024-06-01" => AzureOpenAIClientOptions.ServiceVersion.V2024_06_01,
+                "2024-08-01-preview" => AzureOpenAIClientOptions.ServiceVersion.V2024_08_01_Preview,
+                "2024-10-01-preview" => AzureOpenAIClientOptions.ServiceVersion.V2024_10_01_Preview,
+                _ => throw new NotSupportedException($"The service version '{serviceVersion}' is not supported.")
```

Comment:
If `2024-10-21` came out we'll need to rev this code, is that right?

Any reason not to just pass the version on and have it fail if it's not valid?

Reply:
We can't because the SDK accepts an enum instead of a string and I need some logic to make the conversion, unless instead of exposing a `string` we expose the `SDK type`, which would add breaking changes to API's that were public accepting `string`.

> If 2024-10-21 came out we'll need to rev this code

Yes, every new version or update to the SDK enum needs to be reflected in that function, this might not be a burden as this function is central to all `AzureOpenAI` client logic, we just need to have awareness.